src/lib/sanitize-for-logging.ts

```diff
@@ -28,3 +28,14 @@ export function sanitizeUnstructuredData( data: string ): string {
 		return sanitized.replace( new RegExp( `${ sensitiveKey }.*(\n|$)`, 'ig' ), 'REDACTED\n' );
 	}, data );
 }
+
+// Attempts to sanitize a user path by replacing the user's home directory with a tilde.
+// Returns the original path if it does not match the pattern or if it is not a string.
+export function sanitizeUserpath( path: string ): string {
+	// Matches expected user path on macOS
+	if ( typeof path === 'string' && /\/Users\/[^/]+\/Library/.test( path ) ) {
+		return path.replace( /\/Users\/[^/]+(\/Library)/, '~$1' );
+	}
```

Comment:
Currently, the userpath pattern matcher matches /Users/username/Library/..., which is the expected userpath on macOS (currently, the only supported OS).

When support for other operating systems is added, this userpath matcher can be expanded to cover the new OS userpath patterns.